cmd/kogito/deploy_data_index.go

```diff
@@ -0,0 +1,150 @@
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return deployDataIndexExec(cmd, args)
+		},
+		PreRun:  preRunF,
+		PostRun: posRunF,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(deployDataIndexCmdFlags.infinispanUser) > 0 && len(deployDataIndexCmdFlags.infinispanPassword) == 0 {
+				return fmt.Errorf("Please set infinispan-user and infinispan-password")
+			}
+			if len(deployDataIndexCmdFlags.infinispanUser) == 0 && len(deployDataIndexCmdFlags.infinispanPassword) > 0 {
+				return fmt.Errorf("Please set infinispan-user and infinispan-password")
+			}
+			if len(deployDataIndexCmdFlags.infinispanUser) > 0 &&
+				len(deployDataIndexCmdFlags.infinispanPassword) > 0 &&
+				len(deployDataIndexCmdFlags.infinispanSasl) == 0 {
+				deployDataIndexCmdFlags.infinispanSasl = string(v1alpha1.SASLPlain)
+			}
+			if err := commonCheckDeployArgs(&deployDataIndexCmdFlags.deployCommonFlags); err != nil {
+				return err
+			}
+			if err := commonCheckImageTag(deployDataIndexCmdFlags.image); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+})
+
+var _ = RegisterCommandInit(func() {
+	rootCmd.AddCommand(deployDataIndexCmd)
+	commonAddDeployFlags(deployDataIndexCmd, &deployDataIndexCmdFlags.deployCommonFlags)
+
+	deployDataIndexCmd.Flags().StringVarP(&deployDataIndexCmdFlags.image, "image", "i", resdataindex.DefaultImage, "Image tag (namespace/name:tag) for the runtime Service, e.g: openshift/kogito-data-index:latest")
+	deployDataIndexCmd.Flags().StringVar(&deployDataIndexCmdFlags.kafka.ServiceURI, "kafka-url", "", "The Kafka cluster internal URI, example: my-kafka-cluster:9092")
+	deployDataIndexCmd.Flags().StringVar(&deployDataIndexCmdFlags.infinispan.ServiceURI, "infinispan-url", "", "The Infinispan Server internal URI, example: infinispan-server:11222")
+	deployDataIndexCmd.Flags().StringVar(&deployDataIndexCmdFlags.infinispan.AuthRealm, "infinispan-authrealm", "", "The Infinispan Server Auth Realm for authentication, example: ApplicationRealm")
+	deployDataIndexCmd.Flags().StringVar(&deployDataIndexCmdFlags.infinispanSasl, "infinispan-sasl", "", "The Infinispan Server SASL Mechanism, example: PLAIN")
+	deployDataIndexCmd.Flags().StringVar(&deployDataIndexCmdFlags.infinispanUser, "infinispan-user", "", "The Infinispan Server username")
+	deployDataIndexCmd.Flags().StringVar(&deployDataIndexCmdFlags.infinispanPassword, "infinispan-password", "", "The Infinispan Server password")
+
+	cobra.MarkFlagRequired(deployDataIndexCmd.Flags(), "kafka-url")
+	cobra.MarkFlagRequired(deployDataIndexCmd.Flags(), "infinispan-url")
+})
+
+func deployDataIndexExec(cmd *cobra.Command, args []string) error {
+	var err error
+	if deployDataIndexCmdFlags.project, err = ensureProject(deployDataIndexCmdFlags.project); err != nil {
+		return err
+	}
+	if err := checkKogitoDataIndexCRDExists(); err != nil {
+		return err
+	}
+
+	// if user and password is sent, let's create a secret to hold it and attach to the CRD
```

Comment:
The comment mentions creating a secret, is it meant that secret is created on OpenShift or just as an object in the code? As from code below I cannot see any command actually creating the secret on OpenShift.

Reply:
You're completely right. I forgot to add a one liner to create the secret. Added that plus a unit test to check.

Many thanks!